powersimdata/network/europe_tub/model.py

```diff
@@ -0,0 +1,140 @@
+import os
+import shutil
+from zipfile import ZipFile
+
+from powersimdata.network.constants.region.europe import (
+    abv2country,
+    abv2timezone,
+    interconnect2abv,
+)
+from powersimdata.network.helpers import (
+    check_and_format_interconnect,
+    interconnect_to_name,
+)
+from powersimdata.network.model import ModelImmutables
+from powersimdata.utility.helpers import _check_import
+
+pypsa = _check_import("pypsa")
+zenodo_get = _check_import("zenodo_get")
+
+
+class TUB:
+    """PyPSA Europe network.
+
+    :param str/iterable interconnect: interconnect name(s).
+    :param int reduction: reduction parameter (number of nodes in network). If None,
+        the full network is loaded.
+    :param bool overwrite: the existing dataset is deleted a new dataset is downloaded
+        from zenodo.
+    """
+
+    def __init__(self, interconnect, reduction=None, overwrite=False):
+        """Constructor."""
+        self.grid_model = "europe_tub"
+        self.interconnect = check_and_format_interconnect(
+            interconnect, model=self.grid_model
+        )
+        self.data_loc = os.path.join(os.path.dirname(__file__), "data")
+        self.zenodo_record_id = "3601881"
+        self.reduction = reduction
+
+        if overwrite:
+            self.remove_data()
+
+        self.retrieve_data()
```

Comment:
If the data are already on disk, it will look at the checksum and only download if there is a newer version:
```
>>> from powersimdata.network.europe_tub.model import TUB
>>> tub = TUB("Nordic", reduction=128)
Title: PyPSA-Eur: An Open Optimisation Model of the European Transmission System (Dataset)
Keywords: power system model, capacity expansion model, energy system model, electricity system model, python
Publication date: 2021-09-22
DOI: 10.5281/zenodo.5521712
Total size: 1767.1 MB

Link: https://zenodo.org/api/files/6e58c6dc-083b-46c8-8255-8516e3eb0ed8/networks.zip   size: 1767.1 MB
networks.zip is already downloaded correctly.
All files have been downloaded.
```
We can improve the logic. For example, we don't have to unzip every time we create an object (I tested it and we would gain few ms if we don't do it).

Reply:
Ah, I didn't look into the doc of `zenodo_get`. That's nice.